activerecord/lib/active_record/connection_adapters.rb

```diff
@@ -26,6 +26,10 @@ module ConnectionAdapters
     end
 
     autoload_at "active_record/connection_adapters/abstract/connection_pool" do
+      autoload :ConnectionPool
```

Comment:
ConnectionPool is already on line 41. It seems you only need to move `ConnectionHandler` to that block.

Reply:
👍 Done